doc/page-properties.h

```diff
@@ -0,0 +1,145 @@
+
+\section properties-intro Introduction
+Interface definition files (*.qface) can define properties. See https://github.com/Pelagicore/qface for information
+about QFace interface definition.
+
+In Facelift each property is exposed to the QML engine as Q_PROPERTY with a setter and getter functions, and a change
+signal. If the property is marked as 'readonly' then no setter is generated.
+
+In Facelift properties are generated as members of the &lt;Inerface name&gt;ImplementationBase C++ class.
+The facelift::Property class template provides encapsulation for the internal data. To get the underlying value
+developer can use a getter that returns the const reference to the underlying data (in case of the service property
+it is const reference to a pointer) or one of the specialization specific access functions. The value can be changed only by
+using a setter or one of the specialization specific access functions. This guarantees that the data will not be unnoticibly
+ changed. Each time the setter is called, with a value that differs from the previous one, the change signal is emitted.
+
+\section QML
+
+In case of C++ interface implementation, the property, genetated as a C++ code, is exposed directly to the QML engine
+by Q_PROPERTY macro and can be used without any additional effort.
+
+\section cpp C++
+
+The property in C++ is generated as one of the templated property containers: Property<T>, ListProperty<T>,
+ServiceProperty<T> or ModelProperty<T>. Additionally ListProperty and QMap<QString, T> partial template specializations
+add extra convenience functions: void removeAt(int i), void addElement(ElementType element) and int size() const.
+The QMap<QString, T> specialization is created for the map property defined in *.qface file.
+
+\section subservice Property ownership
+
+The service owns the property objects and maintains their life cycle.
+
+The service property returns a pointer to another service. In this case property object is not an owner of the
+pointed service. In terms of QML engine, the service object ownership is held by C++ code.
+See https://doc.qt.io/qt-5/qqmlengine.html#ObjectOwnership-enum
+
+Other property types are owners of the underyling objects.
+
+
+\section bindings Property bindings
+
+Typically property state change may happen on setter call. An alternative way of updating property value with a minimal
+effort is to bind it to the signal. Whenever property is notified (state change triggered), the property will call
+specified getter function. Getter function must return underlying property type.
+
+In the example below, whenever Context object bound to the property emits signal foo(), the property value is set to
+the value returned from the lambda passed as bind() parameter.
+\code
+
+class Context : public QObject {
+  Q_OBJECT
+public:
+  Q_SIGNAL void foo();
+  facelift::Property<QString> prop;
+};
+
+void bar(Context* ctx){
+  prop.bind([]() {
+          return QString("foo");
```

Comment:
you are returning a constant. Why do you connection a re-evaluation trigger then ? Confusing example !